src/main/java/htsjdk/samtools/BAMRecordCodec.java

```diff
@@ -174,6 +186,18 @@ public void encode(final SAMRecord alignment) {
         }
     }
 
+    /**
+     * Create a "Sentinel" cigar that will be placed in BAM file when the actual cigar has more than 0xffff operator,
+     * which are not supported by the bam format. The actual cigar will be encoded and placed in the CG attribute.
+     * @param cigar actual cigar to create sentinel cigar for
+     * @return sentinel cigar xSyN with readLength (x) and referenceLength (y) matching the input cigar.
+     */
+    public static Cigar makeSentinelCigar(final Cigar cigar) {
+        return new Cigar(Arrays.asList(
+                new CigarElement(cigar.getReadLength(), CigarOperator.S),
+                new CigarElement(cigar.getReferenceLength(), CigarOperator.N)));
+    }
```

Comment:
When writing to BAM, a CIGAR length is at most `(1<<28)-1`. Probably this has been taken care of somewhere else in the code. Just to make sure.

Reply:
where is this restriction coming from?

Reply:
In BAM, the lower 4 bits for the operator and the higher 28 bits for the length, so if either query or the reference length in the alignment longer than `(1<<28)-1`, we will not be able to encode this in BAM. Encoding this in SAM is ok. I guess htsjdk probably has this check already somewhere.

Reply:
From @yfarjoun:

Got it. Did we say that in the spec pr?

I don't know. This limitation is implicit in BAM (like chromosome lengths can't be longer than 2Gbp), but I agree it would be good to make it explicit anyway.

Reply:
This is an extra restriction for the case of a large cigar, not in general..since we will need to put the readlength into the cigar element and there's less room there. The read=length itself is encoded in a 32-bit int. I wonder if we can try to get around this restriction (e.g. if readlength>28-bit use (1<<28-)-1 in the cigar...)

Reply:
In principle, yes. We may do that if Sanger people are happy with it.

Reply:
shouldn't the same restriction be placed on the reference length?